microcosm_flask/conventions/crud_adapter.py

```diff
@@ -59,9 +59,19 @@ def update_batch(self, **kwargs):
 
         """
         items = kwargs.pop("items")
+
+        def transform(item):
+            """
+            Transform the dictionary expected for replace (which uses the URI path's id)
+            into the resource expected from individual resources (which uses plain id).
+
+            """
+            item[self.identifier_key] = item.pop("id")
```

Comment:
The individual resource functions all get called with an argument of the form `foo_id` because this is how we build our URI patterns (e.g. `/api/v1/foo/:foo_id`).

The resources themselves have an `id`; this translates between the two since we're patching with a list of resources and don't have individual URIs to extract the `foo_id` from.